### ISSUE TITLE
Additional API fields

### ISSUE THREAD

**Issue by michaeltorbert** [YahnisElsts/plugin-update-checker#71]
There seem to be additional API fields these days:

contributor (listed on details screen), (author_homepage is author_profile), compatibility (not sure if this is used), ratings can be an array with the number of ratings for each star 1-5, added (not sure if this is used), tags (doesn't seem to be used in details screen), donate link

**Comment by YahnisElsts** [comment]
Most of those were intentionally omitted.

- `compatibility` - specific to wordpress.org. Most other sites and developers don't collect compatibility reports the same way. For most people, `requires` and `tested` are probably enough.
- `ratings` - this also seems specific to WordPress.org. How many commercial developers actually have star ratings on their plugins? Perhaps some of the bigger firms with dozens of plugins might have a rating system, but I can't think of a single example off the top of my head.
- `added` - appears to be unused.
- `tags` - unused, and it wouldn't matter anyway after the user has already installed the plugin.

`donate_link` and `contributors` might be worth adding if there's demand for it.

**Comment by michaeltorbert** [comment]
@YahnisElsts I'd love to see ratings, donate_link, and contributors.
For ratings, devs could either pull info from a .org version of the plugin (for freemium models) or else from our own sites. For instance, we have a plugin store where we sell our own plugins, and potentially other third-party plugin stores may have ratings systems as well.

**Comment by YahnisElsts** [comment]
That's going to be a more complicated than I had assumed. I checked the relevant parts of WP core just now and here's what I found:

- Each rating level links to `https://wordpress.org/support/view/plugin-reviews/`. The URL is hardcoded and I don't see a clean way to replace it. (Messy ways include replacing it with JS, output buffer shenanigans, or rewriting the entire plugin information screen.)
- Contributor names are sanitized with `sanitize_user()`. This includes stuff like converting accented characters to their non-accented variants (which is not so great for non-USA developers like me), removing all HTML tags, etc. Contributor avatars are are loaded from wordpress.org. This URL is also hardcoded and can't be changed. If the contributor name doesn't match a wordpress.org username, you get the default avatar.
- The donation link *only* shows up there are non-zero `contributors`.

Looking at the conditionals in `plugin-install.php`, I could get the donation link to show up by adding a fake contributor that then gets filtered out due to having an empty name and URL. As for the rest, given the restrictions listed above, is it really worth it?
